Add `positron.r.symbols.includeAssignmentsInBlocks` setting

lionel- opened the pull request:
For https://github.com/posit-dev/positron/issues/8330
Extension side of https://github.com/posit-dev/ark/pull/859

Can be merged separately.

lionel- committed:
Add `positron.r.symbols.includeAssignmentsInBlocks` setting

juliasilge reviewed (changes requested):
Thank you so much! 🙌

juliasilge commented on extensions/positron-r/package.json:
```diff
@@ -189,6 +189,11 @@
             "default": true,
             "description": "%r.configuration.diagnostics.enable.description%"
+          },
+          "positron.r.symbols.includeAssignmentsInBlocks": {
```
We chatted elsewhere and cannot come up with a good alternative that will keep us from needing to migrate in the future. So we'll just take our lumps here, and sign up for _more_ future pain. 😭

lionel- commented on extensions/positron-r/package.json:
```diff
@@ -189,6 +189,11 @@
+          },
+          "positron.r.symbols.includeAssignmentsInBlocks": {
+            "type": "boolean",
+            "default": false,
```
Here are some more examples from ellmer.

Numerous local variables in `print.ellmer_prompt()`, some with vague/nondescript names:

<img width="468" height="610" alt="Screenshot 2025-07-25 at 12 48 18" src="https://github.com/user-attachments/assets/16d1aa38-e246-4371-bbb4-c5c2e664eaa3" />

Long file where the general outline is obscured by local variables:

https://github.com/user-attachments/assets/41e47f46-d131-46fb-9c18-326b50b95b9e

Same file with the new setting turned off. It's much easier to see the file structure, i.e. the outline:

https://github.com/user-attachments/assets/d4358127-ddfe-476e-9b6c-26c5e4b1067b

lionel- committed:
Add `workspaceSymbols.includeCommentSections` setting (#8419)

Branched from https://github.com/posit-dev/positron/pull/8412

For https://github.com/posit-dev/positron/issues/4886
Extension side of https://github.com/posit-dev/ark/pull/866

Can be merged separately.
